layouts/_default/single.html

```diff
@@ -6,27 +6,24 @@
   <body class="{{ printf "bg-%s" $bg }}">
     {{ partial "nav.html" . }}
 
-    {{ $page := replace (strings.TrimRight "/" (strings.TrimLeft "/" .RelPermalink)) "/" "__" }}
-
-    {{ with index $.Site.Data $page }}
-      {{ range . }}
-        {{ $.Scratch.Set "data" . }}
-
-        {{ if and (not .disabled) (eq .fragment "hero") }}
-          {{ partial "fragments/hero.html" $ }}
-        {{ end }}
+    {{ range sort .Resources }}
 
+      {{ if and (not .Params.disabled) (eq .Params.fragment "hero") }}
+        {{ partial "fragments/hero.html" .Params }}
       {{ end }}
+
     {{ end }}
 
-    {{ partial "content-single.html" . }}
+    {{ $page := replace (strings.TrimRight "/" (strings.TrimLeft "/" .RelPermalink)) "/" "__" }}
+
+     {{ partial "content-single.html" . }}
 
     {{ with index $.Site.Data $page }}
       {{ range sort . "weight" }}
         {{ $.Scratch.Set "data" . }}
-
-        {{ if and (not .disabled) (eq .fragment "items") }}
-          {{ partial "fragments/items.html" $ }}
+      
+        {{ if and (not .Params.disabled) (eq .Params.fragment "items") }}
+          {{ partial "fragments/items.html" .Params }}
```

Comment:
The changes to call the items partial is correct, but you are working in the old loop. The old loop is looping through the data files, but it needs to loop through the content files (aka range .Resources) as we have used above with hero.

Reply:
Depends on where you are ranging.
A good test is always to print out stuff to check. So before `{{ range sort .Resources }}` just print `{{ .Resources }}` to see, if any pages are actually available.
My assumption (haven't seen the latest code) is that you switched from `{{ range sort . "weight" }}` to `{{ range sort .Resources }}` this would still run in the context of `{{ with index $.Site.Data $page }}`.
Instead you can move this call for the items partial to below the call for the hero one. This is then in the correct loop.
Within a PR you can also push temporary commits and before we merge, we can rebase and cleanup these intermediate commits. Makes it easier to review changes and less likely I am making horrible assumptions.